src/Subject.ts

```diff
@@ -123,6 +123,12 @@ export class Subject<T> extends Observable<T> implements Observer<T>, Subscripti
     this._complete();
   }
 
+  asObservable(): Observable<T> {
+    return new Observable((observer: Observer<any>) => {
```

Comment:
Thought about casting too a bit,but wanted tobstay in type guarded world ;) think derived class would work in this case. Thanks!